lib/fortnox/api/repositories/base/pagination.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Fortnox
+  module API
+    module Repository
+      module Pagination
```

Comment:
I think we should drop this and create a proper class for the metadata if we want methods on it instead.

Right now the @metadata is an instance of the metadata model, that is fine. But the pagination shouldn't live on the repository singleton. The way me current example looks it keeps all that in the single call. Your implementation leaves things on the global repository object instead ....
A better place for it would be the collection we are building, `results = []` in my example. By creating a collection class and pushing the logic to that we could handle pagination in that and support things like batching from the API etc in that object instead.
But for now I think it is ok to just stick it in the main method. Check this for more on why I think that: https://github.com/my-codeworks/fortnox-api/issues/149 (base gem work can be found at https://github.com/Accodeing/rest_easy)